lib/heuristics/periodic_heuristic.rb

```diff
@@ -703,29 +703,31 @@ def insertion_cost_with_tw(timewindow, route_data, service, position)
       [activity_out, tw_accepted, shift, time_back_to_depot]
     end
 
+    # For each day where at least one visit was recently assigned to vehicle_id,
+    # insert minimum required visits in order to ensure that route will not be emptied later
     def ensure_routes_will_not_be_rejected(vehicle_id, impacted_days)
       while impacted_days.size.positive?
         started_day = impacted_days.first
-        can_not_insert_more = false
         route_cost_fixed = @candidate_routes[vehicle_id][started_day][:cost_fixed]
         route_exclusion_costs =
           @candidate_routes[vehicle_id][started_day][:stops].map{ |stop|
             @services_data[stop[:id]][:raw].exclusion_cost || 0
           }.reduce(&:+)
-        while route_exclusion_costs < route_cost_fixed || can_not_insert_more
+        while route_exclusion_costs < route_cost_fixed
           inserted_id, _unlocked_ids = try_to_add_new_point(vehicle_id, started_day)
           impacted_days |= adjust_candidate_routes(vehicle_id, started_day)
           if inserted_id
             route_exclusion_costs += @services_data[inserted_id][:raw].exclusion_cost.to_f
           else
-            can_not_insert_more = true
+            break
```

Comment:
Next time can you separate the actual fix and the other stuff like the syntax and comment? Here, I accept that it is quite manageable but if we don't have the discipline to separate commits, PRs get nasty quite quickly ;)